server/pulp/server/controllers/repository.py

```diff
@@ -105,14 +105,20 @@ def find_repo_content_units(
             yield_count += 1
 
 
-def find_units_not_downloaded(repository):
+def _get_paged_repo_unit_ids_with_model(repository):
```

Comment:
I have something similar going in a branch right now that might be more reusable. Let's compare notes tomorrow.